### PR TITLE
WIP updated wallet manager interfaces to support multiple wallets

### PR DESCRIPTION
**Pull request by rweald**
Dev can choose whether they want multiple mnemonics, or multiple
keypairs from 1 mnemonic.

Added some extra code level docs to start describing our interfaces and
the params more.

### CODE REVIEW

**Comment by rweald** [comment]
This PR does not yet actually work through the stack, it's just trying to get a feel for the interface changes, before we do the work to update the underlying native code to support our objective.

File: lib/wallet_manager.dart [hunk 1]
```diff
@@ -31,13 +47,23 @@ class WalletManager {
 
     final mnemonic = await _keyManager.generateMnemonic();
```
Comment (DonGambas) [inline lib/wallet_manager.dart:48]: add identifier here

File: lib/wallet_manager.dart [hunk 2]
```diff
@@ -19,8 +19,24 @@ class WalletManager {
     return _instance;
   }
 
+  /*
+    * Creates a new wallet and saves it to the device.
+    * If a wallet already exists, it will throw an error unless overwrite is set to true.
+    *
+    * overwrite: If true, will overwrite the existing wallet. Use with caution as old mnemonic will be lost.
+    * identifier: The identifier for the underlying mnemonic. You should provide a value here is you intend to use multiple unique mnemonics.
```
Comment (DonGambas) [inline lib/wallet_manager.dart:27]: should provide value here *if* you intend

**Comment by rweald** [comment]
I'm going to close this for now. Was a good spike to see what changes we'd need, but right now this new feature isn't high enough on the priority list for the complexity.

**Pull request closed without merging**